scipy/sparse/csgraph/parameters.pxi

```diff
@@ -10,6 +10,11 @@ ctypedef fused int32_or_int64:
     np.int32_t
     np.int64_t
 
+# Another copy of the same fused type, for working with mixed-type functions.
+ctypedef fused int32_or_int64_b:
+    np.int32_t
+    np.int64_t
```

Comment:
If we want to use `const`-qualification, we need this work-around for https://github.com/cython/cython/issues/5230 (see [the original types aliasing](https://github.com/cython/cython/blob/master/Cython/Includes/numpy/__init__.pxd#L315-L320)).

```suggestion
    signed int
    signed long long

# Another copy of the same fused type, for working with mixed-type functions.
ctypedef fused int32_or_int64_b:
    signed int
    signed long long
```

If this does not work, we can proceed in two steps like scikit-learn's:
 - [individual type definition](https://github.com/scikit-learn/scikit-learn/blob/04860335c82d557e663b4cfa218663d1c7bf65fd/sklearn/utils/_typedefs.pxd#L26-L27)
 - [fused type definition](https://github.com/scikit-learn/scikit-learn/blob/04860335c82d557e663b4cfa218663d1c7bf65fd/sklearn/utils/_typedefs.pyx#L9-L15)

Reply:
Can we not get away with just Py_ssize_t ?

Reply:
If this allows treating 32bit and 64bit integer without copy, then yes.

Reply:
To my understanding, `Py_ssize_t` means "int32 or int64, depending on your platform", but what we need here is support for both int32 _and_ int64 in the same function (unfortunately).